core-tests/src/test/java/org/mule/MuleSessionHandlerTestCase.java

```diff
@@ -87,25 +90,24 @@ public static void restoreEncoding()
     @Test
     public void testSessionProperties() throws Exception
     {
-        DefaultMuleMessage message = new DefaultMuleMessage("Test Message", muleContext);
+        MuleEvent event = MuleTestUtils.getTestEvent("Test Message", muleContext);
         SessionHandler handler = new SerializeAndEncodeSessionHandler();
-        MuleSession session = new DefaultMuleSession();
 
         String string = "bar";
-        session.setProperty("fooString", string);
+        event.getSession().setProperty("fooString", string);
 
         Date date = new Date(0);
-        session.setProperty("fooDate", date);
+        event.getSession().setProperty("fooDate", date);
 
         List<String> list = createList();
-        session.setProperty("fooList", list);
+        event.getSession().setProperty("fooList", list);
 
-        handler.storeSessionInfoToMessage(session, message);
+        handler.storeSessionInfoToMessage(event.getSession(), event.getMessage());
         // store save session to outbound, move it to the inbound
         // for retrieve to deserialize
-        Object s = removeProperty(message);
-        message.setInboundProperty(MuleProperties.MULE_SESSION_PROPERTY, s);
-        session = handler.retrieveSessionInfoFromMessage(message);
+        Object s = removeProperty(event);
+        ((DefaultMuleMessage) event.getMessage()).setInboundProperty(MuleProperties.MULE_SESSION_PROPERTY, s);
+        MuleSession session = handler.retrieveSessionInfoFromMessage(event.getMessage());
```

Comment:
I'm not sure about this: session lives in the event, but the session handler is reading/writing from/to the message. Is that right?